compiler/coreSyn/CoreLint.hs

```diff
@@ -2463,12 +2507,14 @@ addAliasUE id ue thing_inside = LintM $ \ env errs ->
   in
     unLintM thing_inside (env { le_ue_aliases = new_ue_aliases }) errs
 
+addAliasUEs :: [(Id, UsageEnv)] -> LintM a -> LintM a
+addAliasUEs bnds thing_inside =
+  foldl (flip . uncurry $ addAliasUE) thing_inside bnds
+
 varCallSiteUsage :: Id -> LintM UsageEnv
 varCallSiteUsage id =
   do m <- getUEAliases
-     return $ case lookupNameEnv m (getName id) of
-         Nothing -> unitUE id One
-         Just id_ue -> id_ue
+     return $ varCallUsage m id
```

Comment:
Hm. So it seems that you are using the aliases at usages. But then again at checking? I think I'm missing something here.